analyzers/src/SonarAnalyzer.CSharp/Rules/MethodParameterMissingOptionalCodeFix.cs

```diff
@@ -60,26 +61,22 @@ protected override async Task RegisterCodeFixesAsync(SyntaxNode root, CodeFixCon
             context.RegisterCodeFix(
                 CodeAction.Create(
                     Title,
-                    c =>
+                    _ =>
```

Comment:
This is not a discard. It is a variable with a less clear name `_` that can be used to do `_.ThrowIfCancellationRequested();`. I prefer the old name.
```suggestion
                    c =>
```

Reply:
It tells that the variable is not used. It's a common used pattern.

Reply:
I've seen that pattern required on method parameters too by the community. And it was not a good idea to have public `Method(int _)`

Reply:
I don't understand why. It makes it obvious that the parameter is not used and points out that maybe a refactoring is needed.

Reply:
We can't change the source of the signature anyway. None of the variant (`_`, `c`) is used in the code. The difference is that we loose information about what is available.